### PR TITLE
 Implement Classroom, Book, and Rental Classes with Relationships

### PR DESCRIPTION
**Pull request by MussieTeka**
**Classroom Class:**

- Created `Classroom` class with `@label` instance variable initialized in the constructor.
- Implemented setter and getter methods for `@label` using `attr_accessor`.
- Established has-many/belongs-to relationship between `Classroom` and `Student`:
  - Implemented has-many side (classroom has many students).
  - Implemented belongs-to side (student belongs to a classroom).
  - Ensured that adding a student to a classroom sets the classroom for the student.
  - Ensured that setting the classroom for a student also adds the student to the classroom's students.

**Book Class:**

- Created `Book` class with `@title` and `@author` instance variables initialized in the constructor.
- Implemented setter and getter methods for `@title` and `@author` using `attr_accessor`.

**Rental Class:**

- Created `Rental` class with `@date` instance variable initialized in the constructor.
- Implemented setter and getter methods for `@date` using `attr_accessor`.
- Established many-to-many relationship between `Person` and `Book` using `Rental`:
  - Implemented has-many side of Book and Rental (book has many rentals).
  - Implemented belongs-to side of Rental and Book (rental belongs to a book).
  - Implemented has-many side of Person and Rental (person has many rentals).
  - Implemented belongs-to side of Rental and Person (rental belongs to a person).
  - Modified constructor of Rental to set associated Book and Person.

The changes have been thoroughly tested and meet the specified requirements. Thank you for reviewing my work.

### CODE REVIEW

File: person.rb [hunk 1]
```diff
@@ -2,14 +2,15 @@
 
 class Person < Nameable
   attr_accessor :name, :age
-  attr_reader :id
+  attr_reader :id, :rentals
```
Comment (Ismailco) [inline person.rb:5]: - [ ] Same here kindly consider adding `rentals` to the `attr_accessor` instead of just `attr_reader`

File: person.rb [hunk 2]
```diff
@@ -2,14 +2,15 @@
...
 
   def initialize(id, name = 'Unknown', age = nil, parent_permission: true)
```
Comment (Ismailco) [inline person.rb:7]: - [ ] The id should be generated automatically, kindly consider removing it from the constructor, also age doesn't have a default value kindly make sure to remove `= nil`

File: person.rb [hunk 3]
```diff
@@ -20,6 +21,10 @@ def correct_name
     name
   end
 
+  def add_rental(rental)
+    rentals << rental
+  end
```
Comment (Ismailco) [inline person.rb:26]: - [ ] Same here for the `add_rentals` method it should use the `Rental` class to create a new rental. Your code should look something like this  👇🏼 
```ruby
def add_rental(book, date) 
  Rental.new(date, book, self)
end
```

**Review by Ismailco — CHANGES_REQUESTED** [review]
### Hi @MussieTeka, 

Good job so far 👍🏼 
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights

✔️ No linter errors
✔️ Used Gitflow
✔️ Work documented professionally

## Required Changes ♻️
- [ ] Kindly make sure to add getters `attr_reader` for the `classroom` variable in the `Student` class.
- [ ] Also make sure a method that serves as the 'belongs-to' side is present in the `Student` class, e.g `def classroom=(classroom).....`

_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: classroom.rb [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+  attr_accessor :label
+  attr_reader :students
+
+  def initialize(label)
+    @label = label
+    @students = []
+  end
+
+  def add_student(student)
+    students << student
```
[the remark this answers is not in the document]
Comment (MussieTeka) [reply classroom.rb:11]: Thank you for the feedback @Ismailco, I have added the missing `@` before the `students` variable.

File: person.rb (hunk 3 above)
Comment (MussieTeka) [reply person.rb]: Thank you for your feedback, @Ismailco. I have revised the entire codebase according to the project requirements.

File: person.rb (hunk 2 above)
Comment (MussieTeka) [reply person.rb]: Thank you for your feedback, @Ismailco. I have revised the entire codebase according to the project requirements.

File: person.rb (hunk 1 above)
Comment (MussieTeka) [reply person.rb]: Thank you for your feedback, @Ismailco. I have revised the entire codebase according to the project requirements.

File: rental.rb [hunk 5]
```diff
@@ -0,0 +1,12 @@
+class Rental
+  attr_accessor :date
+  attr_reader :book, :person
+
+  def initialize(date, book, person)
+    @date = date
+    @book = book
+    @person = person
+    book.add_rental(self)
+    person.add_rental(self)
```
[the remark this answers is not in the document]
Comment (MussieTeka) [reply rental.rb]: Thank you for your feedback, @Ismailco. I have revised the entire codebase according to the project requirements.

File: rental.rb [hunk 6]
```diff
@@ -0,0 +1,12 @@
+class Rental
+  attr_accessor :date
+  attr_reader :book, :person
```
[the remark this answers is not in the document]
Comment (MussieTeka) [reply rental.rb]: Thank you for your feedback, @Ismailco. I have revised the entire codebase according to the project requirements.

File: book.rb [hunk 7]
```diff
@@ -0,0 +1,14 @@
...
+
+  def initialize(title, author)
+    @title = title
+    @author = author
+    @rentals = []
+  end
+
+  def add_rental(rental)
+    rentals << rental
+  end
```
[the remark this answers is not in the document]
Comment (MussieTeka) [reply book.rb]: Thank you for your feedback, @Ismailco. I have revised the entire codebase according to the project requirements.

File: book.rb [hunk 8]
```diff
@@ -0,0 +1,14 @@
+class Book
+  attr_accessor :title, :author
+  attr_reader :rentals
```
[the remark this answers is not in the document]
Comment (MussieTeka) [reply book.rb]: Thank you for your feedback, @Ismailco. I have revised the entire codebase according to the project requirements.

**Review by alexander16108 — APPROVED** [review]
Hi @MussieTeka , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights
✔️ No linter errors
✔️ Used Gitflow
✔️ Work documented professionally

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**